source/NVDAObjects/window/winConsole.py

```diff
@@ -26,6 +25,18 @@ def initOverlayClass(self):
 		if isinstance(self, KeyboardHandlerBasedTypedCharSupport):
 			self._supportsTextChange = False
 
+	def _get_diffAlgo(self):
+		# Non-enhanced legacy consoles use caret proximity to detect
+		# typed/deleted text.
+		# Single-character changes are not reported as
+		# they are confused for typed characters.
+		# Force difflib to keep meaningful edit reporting in these consoles.
+		if not isinstance(self, KeyboardHandlerBasedTypedCharSupport):
+			from diffHandler import get_difflib_algo
+			return get_difflib_algo()

```

Comment:
Please move this to `KeyboardHandlerBasedTypedCharSupport` and (along with the comment). Logic dependent on type like this indicates a problem with the design.

Reply:
This is a particularity of the legacy console code: some legacy consoles subclass `KeyboardHandlerBasedTypedCharSupport`, and some don't. We want to allow DMP on all objects, except legacy consoles that aren't enhanced.

Reply:
Those particular subclasses that shouldn't use DMP should then override the property. Now this might be tricky because of dynamically added classes to the class list. If necessary you can add a class for these "legacy consoles that aren't enhanced" that overrides the diff algorithm.